docs/design/ttl_scans.md

```diff
@@ -0,0 +1,31 @@
+# TTL scans
+
+## Summary
+
+Add an option to automatically delete old starboard scans, in this first version focus on vulnerability scans but in the long run we could add similar functionality to other reports as well.
+
+## Motivation
+
+In [537](https://github.com/aquasecurity/starboard/issues/537) we talk about a need to run nightly vulnerability scans of CVE:s.
+This way we can make sure to get new CVE reports for long time running pods as well.
+
+## Proposal
+
+Add a feature feature flag of the operator, for example `--ttlVulnerabilityReport 24h`, this way we can add other ttl values for other reports as well.
+Adding this flag would add a spec/annotation to the generated VulnerabilityReport that we can look for.
+
+Using similar functionality as defined in [ttlAfterFinished](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) for cronjob seems to be the most versatile option.
+
+Just like the [ttlAfterFinished controller](https://github.com/kubernetes/kubernetes/blob/2a821d787b7f21fbb399b0a0bf6d3fa0c79f4293/pkg/controller/ttlafterfinished/ttlafterfinished_controller.go) we should add it as a separate controller.
+This way it will be easy to extend to other reports as well.
+
+## Alternatives
+
+Another "simpler" option could be to add the same feature flag `--ttlVulnerabilityReport 24h` but instead of doing a similar solution as defined in ttlAfterFinished we could create a cronjob/job that runs once an hour and check for the same annotation.
```

Comment:
I really like the "event driven" approach described in the proposal. Especially to avoid deleting multiple VulnerabilityReports around the same time, which would create multiple (re)scan Jobs. Not sure how feasible is that, but more jitter in delete events the better overall.

With a CronJob we'd need to create it programmatically or add it as part of the installation descriptors. We'd also need a container and command to run it.

Reply:
That may or may not be seen as a good thing depending on the user.

Some users might want to know that starting at some point in time (daily, weekly, whatever) that all scans have happened recently.

The operator one way or another will need to be stable in the face of a lot of new scan requests, so potentially triggering a lot of rescans needs to be handled anyway I think.

I guess the event based one (24h + age of object) can still be pretty easily be turned into the latter by just hard deleting all the reports with a cron job and skipping the TTL feature by an end user. So they are not mutually exclusive.

Reply:
Let's see how the event drive approach is adapter. We can always think of enhancement and add CronJob if it's really needed.